lib/crowbar/init/application.rb

```diff
@@ -471,13 +493,17 @@ class Application < Sinatra::Base
 
           if init[:body] # nil body means success
             logger.error(init[:body])
-            res[:error] = init[:body]
+            res[:errors] = {
+                data: init[:body]
+            }
             http_code = 422
           end
         end
 
-        if result[:error]
-          upgrade_status.end_step(false, database: result[:error])
+        if result[:errors] || http_code != 200
+          # reformat the error response to match common format
+          result[:errors] = { database_connect: result[:errors] }
```

Comment:
Just to be sure we don't miss any error.
I hit the `nil` error message multiple times (need to do more debugging to figure out what's going on).
If we just check the `result[:errors]` we can get success response and `end_step` even if there was an error.

I was thinking about just checking `http_code` but there can be some change in the future which will result in 200 with non-empty error. Just to be sure we don't miss this and to make the check more explicit (if the condition contains word "errors" it must be about errors, right?) I decided to keep both parts.